mod_authnz_pam.c

```diff
@@ -32,6 +32,8 @@
 
 #include "mod_auth.h"
 
+module AP_MODULE_DECLARE_DATA authnz_pam_module;
```

Comment:
This already exists on line 360.

Reply:
I missed that one on line 169, but afaik it has to be the first thing after the includes. So we should delete the one on 169 and keep the one I added.

You can test it, but the ifdef will basically always be `AP_DECLARE_MODULE` unless you use an ancient Apache version, yet apxs still wasn't able to find the module name even with line 169. But if you move it to the beginning, it works.

I've removed line 169 and pushed another commit. I am way past my bedtime, so I'm heading to bed now.

If you play around with my changes, you will see that `apxs -i` works now.

Reply:
> afaik it has to be the first thing after the includes

It does not. Why should it? It's just a C.

I mean, it does not hurt to move it up but I'm concerned that you use wrong rationales.

> yet apxs still wasn't able to find the module name even with line 169

It's because the regexp
```
$content =~ m|.*AP_DECLARE_MODULE\s*\(\s*([a-zA-Z0-9_]+)\s*\)\s*=.*|s || $content =~ m|.*module\s+(?:AP_MODULE_DECLARE_DATA\s+)?([a-zA-Z0-9_]+)_module\s*=\s*.*|s
```
in https://github.com/apache/httpd/blob/trunk/support/apxs.in#L534 looks for an equal sign after the `AP_DECLARE_MODULE()` so it is the intervening `#else ... #endif` that stops it from working. It has nothing to do with the declaration on line 169 or anywhere you might move it -- that `module AP_MODULE_DECLARE_DATA authnz_pam_module;` is not what matches, no matter where you put it.

> But if you move it to the beginning, it works

It does not, unless you remove the `#ifdef ... #endif` construct as well.

Reply:
> I've removed line 169 and pushed another commit.

Thanks. Can you please squash the commits to one and use some more descriptive commit message, something like "Addressing apxs:Error: Sorry, cannot determine bootstrap symbol name with apxs -i."?

Reply:
> I mean, it does not hurt to move it up but I'm concerned that you use wrong rationales.

Yes, you are right. I came across a few weird compiler issues and optimization errors a while back and it was late, I mixed up few things. My bad.

> looks for an equal sign after the AP_DECLARE_MODULE

Yep, I saw that when I clicked the link to your PR for apxs. I read it after I replied in this thread. ;-)

> Can you please squash the commits to one and use some more descriptive commit message

Will do in a bit.